internal/appstore/appstore.go

```diff
@@ -157,29 +157,21 @@ func (s *AppStore) Find(filename string) (storage.ArtifactMeta, error) {
 		return storage.ArtifactMeta{}, err
 	}
 
-	queryString := fmt.Sprintf("?name=%s", filepath.Base(filename))
-	for {
-		request, err := createLocateRequest(fmt.Sprintf("%s/v1/storage/list", s.URL), s.Username, s.AccessKey, queryString)
-		if err != nil {
-			return storage.ArtifactMeta{}, err
-		}
-
-		lr, err := s.executeLocateRequest(request)
-		if err != nil {
-			return storage.ArtifactMeta{}, err
-		}
-
-		for _, item := range lr.Items {
-			if item.ETag == hash {
-				return storage.ArtifactMeta{ID: item.ID}, nil
-			}
-		}
+	queryString := fmt.Sprintf("?sha256=%s", hash)
+	request, err := createLocateRequest(fmt.Sprintf("%s/v1/storage/list", s.URL), s.Username, s.AccessKey, queryString)
```

Comment:
please avoid using /list as it is going to be deprecated. Prefer using `/files` instead

Reply:
updated.